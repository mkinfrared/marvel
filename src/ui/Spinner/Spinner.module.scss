@import "styles/colors";

.Spinner {
  border: 5px solid;
  border-color: $primary-main;
  border-radius: 50%;
  height: 25px;
  position: relative;
  width: 25px;

  &:before {
    animation: animate 1s linear infinite;
    border-bottom: 5px solid $secondary-main;
    border-left: 5px solid transparent;
    border-radius: 50%;
    border-right: 5px solid transparent;
    border-top: 5px solid $secondary-main;
    content: "";
    height: 100%;
    left: -5px;
    position: absolute;
    top: -5px;
    width: 100%;
  }
}

@keyframes animate {

  0% {
    transform: rotate(0deg);
  }

  25% {
    transform: rotate(180deg);
  }

  50% {
    transform: rotate(360deg);
  }

  75% {
    transform: rotate(540deg);
  }

  100% {
    transform: rotate(720deg);
  }
}
