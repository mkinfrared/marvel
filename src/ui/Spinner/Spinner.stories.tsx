import { Meta, Story } from "@storybook/react/types-6-0";
import React from "react";

import { Spinner } from "./Spinner";

export default {
  title: "UI/Spinner",
  component: Spinner
} as Meta;

const Template: Story = () => (
  <div style={{ padding: 20 }}>
    <div>
      <Spinner />
    </div>
    <div style={{ marginTop: 20 }}>
      <Spinner size={2} />
    </div>
  </div>
);

const Primary = Template.bind({});

Primary.args = {};

export { Primary };
