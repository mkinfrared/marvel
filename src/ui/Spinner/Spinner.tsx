import React from "react";

import css from "./Spinner.module.scss";
import { SpinnerProps } from "./Spinner.type";

const Spinner = ({ size = 1 }: SpinnerProps) => {
  const side = 25 * size;

  const style = {
    width: side,
    height: side
  };

  return <div className={css.Spinner} style={style} />;
};

export { Spinner };

export default React.memo(Spinner);
