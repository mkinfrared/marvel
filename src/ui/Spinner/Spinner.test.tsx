import { render } from "@testing-library/react";
import React from "react";

import { Spinner } from "./Spinner";

describe("<Spinner />", () => {
  const Component = Spinner;

  it("should be defined", () => {
    const { container } = render(<Component />);

    expect(container).toBeDefined();
  });

  it("match the snapshot", () => {
    const { container } = render(<Component />);

    expect(container).toMatchSnapshot();
  });
});
