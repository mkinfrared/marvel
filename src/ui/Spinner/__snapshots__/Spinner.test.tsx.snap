// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> match the snapshot 1`] = `
<div>
  <div
    class="Spinner"
    style="width: 25px; height: 25px;"
  />
</div>
`;
